libs/payments/cart/src/lib/cart.service.ts

```diff
@@ -636,13 +636,25 @@ export class CartService {
           cart.stripeSubscriptionId
         );
       }
+
+      let paymentMethodType: string;
+      if (this.subscriptionManager.getPaymentProvider(subscription) === 'paypal') {
+        paymentMethodType = SubPlatPaymentMethodType.PayPal;
+      } else {
+        const stripePaymentMethod = await this.paymentMethodManager.retrieve(
+          subscription.default_payment_method ?? '',
+        );
+        paymentMethodType = stripePaymentMethod.type as string;
+      }
+
       await this.checkoutService.postPaySteps({
         cart,
         version: cart.version,
         subscription,
         uid: cart.uid,
         paymentProvider:
           this.subscriptionManager.getPaymentProvider(subscription),
+        paymentForm: paymentMethodType as SubPlatPaymentMethodType     // check
```

Comment:
This typecast should be removed. Create a utility to convert Stripe.PaymentMethod.Type to the enum using a dictionary.